core-libs/setup/ssr/logger/loggers/w3c-trace-context/parse-traceparent.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { InvalidTraceparentFormatError } from './errors/invalid-traceparent-format-error';
+import { InvalidTraceparentLengthError } from './errors/invalid-traceparent-length-error';
+import { W3cTraceContext } from './w3c-trace-context.model';
+
+const HEXDIGLC = '[0-9a-f]'; // https://www.w3.org/TR/trace-context/#traceparent-header-field-values
+const VERSION = HEXDIGLC + '{2}'; // 2 HEXDIGLC
+const TRACE_ID = HEXDIGLC + '{32}'; // 32 HEXDIGLC
+const PARENT_ID = HEXDIGLC + '{16}'; // 16 HEXDIGLC
+const TRACE_FLAGS = HEXDIGLC + '{2}'; // 2 HEXDIGLC
+const TRACEPARENT = [VERSION, TRACE_ID, PARENT_ID, TRACE_FLAGS].join('-'); // separated by dashes
+const traceparentPattern = new RegExp('^' + TRACEPARENT + '$');
+
+/**
+ * Maps traceparent header to object with properties version, traceId, spanId and traceFlags.
+ *
+ * @param traceparent
+ * @returns Params of the traceparent header
+ *
+ * @see https://www.w3.org/TR/trace-context/#traceparent-header-field-values
+ */
+export function parseTraceparent(
+  traceparent: string | undefined | null
+): W3cTraceContext | null {
+  if (typeof traceparent !== 'string') {
+    return null;
+  }
```

Comment:
For consistency, did you consider throwing an error e.g. `InvalidTraceparentTypeError` instead of returning null?

IMHO it looks more robust to have return type of `parseTransparent` only `W3cTraceContext` (without  `... | null`).

What do you think?

Reply:
What about the case where `traceparent` is `null` or  `undefined`? I don't want to throw an error if the value is missing. Because an input type can be `string | null | undefined`, the `if statement` mentioned above seems to be a type-safe condition that checks whether `traceparent` is defined or not.